example/30-infrastructure.yaml

```diff
@@ -62,5 +62,6 @@ spec:
     networks:
     # router:
     #   id: 1234
+    #   floatingPoolSubnetId: my-floating-pool-subnet-id
```

Comment:
Why didn't you put it next to `floatingPoolName` but below `router`?

Reply:
Hm, but the `floatingPoolName` is also a property of the router. https://github.com/gardener/gardener-extension-provider-openstack/pull/92/files#diff-f8129006b7b8dad2ed592a618872d49fR23
Also, I find it confusing that we use "name" for the floating pool but "id" for the floating pool subnet.

Reply:
Having the name instead of the idea is a great proposal and we can fetch the corresponding id probably via a data source. I'll do that.

For the structure of the `router` and the `floatingPoolName` I struggle a bit. IMO both belongs to the `router` section, but moving the `floatingPoolName` field to the `router` section would require some migration logic. So what would you suggest? Moving the the `subnetName` to same level as the `floatingPoolName` or the other way around? 

cc @MartinWeindel

Reply:
I would probably add the `floatingPoolSubnetName` to the same level as `floatingPoolName` now, and strive for an improved structure by moving both to `router` in the next API version. But I have no strong opinion. If you think it's okay as it's now then fine.

Reply:
Sound like a great proposal. I will adapt the PR accordingly and open an issue to restructure the `router` section in the infraconfig for next api-version.